lib/scrapPaperReader.js

```diff
@@ -10,4 +85,410 @@ const readPdf = async (pdfBuffer) => {
+    logger.info(`JSON decoder successfully read barcodes on pages: ${readBarcodedPages.length}`);
+    logger.info(`JSON decoder all results data dump: ${JSON.stringify(allResults, null, 2)}`);
+
+    // get iq details, ai details to store in file API
+    const barcodes = readBarcodedPages.map((page) => page.barcode);
+    const query = sql.get_barcode_metadata.replace(
+      '$match',
+      `s.submitted_answer->>'_pdf_barcode_scan' = '${barcodes.join(
+        "' OR s.submitted_answer->>'_pdf_barcode_scan' = '"
+      )}'`
+    );
+    const fileMetadata = await sqldb.queryAsync(query, {});
+
+    const uploadedFiles = await _uploadPages(readBarcodedPages, fileMetadata, userId);
+    await _updateBarcodesTable(uploadedFiles);
+
+    logger.info(`Removing working directory: ${workingDir}`);
+    await removeFilesAndDir(workingDir, logger);
+  } catch (err) {
+    logger.info(`Operation aborted due to error: ${err.message}`);
+    logger.info(`Removing working directory: ${workingDir}`);
+
+    await removeFilesAndDir(workingDir, logger);
+
+    logger.fail(
+      // If we want the front-end to see this debugging information
+      `
+      ${err.message}
+      ${err.stack}
+      `
+    );
+  }
+  logger.succeed();
+};
+
+/**
+ * A Pdf Scan is a collection of scanned barcoded single pieces of paper with
+ * student written work. It is expected that one barcode exists on each page in the PDF.
+ * If a barcode is readable by the decoder, the page will be uploaded to S3 and stored with
+ * metadata information in `files` and `barcodes` table to later allow viewing of doc by student.
+ * If a student submits a barcode after file upload, it can still be displayed on front-end.
+ * @param {Buffer} pdfBuffer buffered application/pdf scanned document
+ * @param {string} originalName uploaded filename
+ * @param {string} userId authn user id to be stored in file api
+ * @param {object} logger
+ * @returns updates barcode table rows
+ */
+const processScrapPaperPdf = async (pdfBuffer, originalName, userId, logger) => {
+  const job = async () => _processScrapPaperPdf(pdfBuffer, originalName, userId, logger);
+  Queue.enqueue(job, logger);
+};
+
+const _decodeJpegs = async (jpegs, logger) => {
+  for (let i = 0; i < jpegs.length; i++) {
+    const start = new Date();
+    logger.info(`_decodeJpegs() - start decoding of ${jpegs[i].jpegFilename} / page ${i}`);
+    jpegs[i]['barcode'] = await _decodeJpeg(jpegs[i], logger);
+    const secondsElapsed = (new Date() - start) / 1000;
+    logger.info(
+      `_decodeJpegs() - took ${secondsElapsed} to decode ${jpegs[i].jpegFilename} / page ${i}`
+    );
+  }
+  return jpegs;
+};
+
+/**
+ * Detects and extracts a barcode on a jpeg file
+ * We probably want to start working on some image analysis that helps read
+ * barcodes based on uploaded scans. Many likely will not read. This will require
+ * some optimization here.
+ * @param {string} jpeg jpeg file object from arr produced by _convertPdf()
```

Comment:
I won't leave comments on every function, but: if you're going to include type annotations, please ensure that they a) include all parameters and b) correctly reflect the type of the parameters. For instance, this function is missing a `logger` param and `jpeg` is most definitely not a string.

Reply:
That really goes without saying. It is almost as if there is more than meets the eye here. 

Did you want the comments out then? Just let me know what you prefer.

Reply:
I am a huge proponent of type annotations in comments, I just want them to actually reflect what the types are. Otherwise that creates unnecessary cognitive overhead to understanding the code *and* makes our work more difficult when this code is ultimately type-checked in the future.